src/PrestaShopBundle/Resources/config/services/bundle/form/form_factory.yml

```diff
@@ -0,0 +1,10 @@
+services:
+  _defaults:
+    public: true
+
+  prestashop.bundle.form.admin.improve.design.theme.page_layout_customization_form_factory:
+    class: 'PrestaShopBundle\Form\Admin\Improve\Design\Theme\PageLayoutCustomizationFormFactory'
+    arguments:
+      - '@form.factory'
+      - '@prestashop.core.addon.theme.repository'
+      - '@=service("prestashop.adapter.legacy.context").getContext().shop.theme.getName()'
```

Comment:
Same here, I'm concerned about these types of calls as this is dynamic and can change